File: 백준/Silver/9020. 골드바흐의 추측/골드바흐의 추측.py
def isPrime(num):
    if num==1:
        return False
    else:
        for i in range(2, int(num**0.5)+1):
            if num%i == 0:
                return False
        return True
    
num = int(input())
for i in range(num):
    a  = []
    n = int(input())
    
    for j in range(n//2, 0, -1):
        if isPrime(j) and isPrime(n-j):
            print(j, n-j)
            break

File: 백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        boolean[] isPrime = new boolean[1000001];
        for (int i = 2; i <= isPrime.length-1 ; i++) {
            isPrime[i] = true;
        }
        for(int i =2; i <= isPrime.length -1; i++){
            for(int j = i*2; j<=isPrime.length -1; j += i){
                    if(!isPrime[j]){
                        continue;
                    }
                    isPrime[j] = false;
            }
        }
        while(true){
            int n = Integer.parseInt(br.readLine());
            boolean num = false;
            if(n == 0){
                break;
            }
            for(int i = 2; i <= n/2; i++){
                if(isPrime[i] && isPrime[n-i]){
                    System.out.println(n + " = " + i + " + " + (n-i));
                    num = true;
                    break;
                }
            }
            if(!num){
                System.out.println("Goldbach's conjecture is wrong.");
            }
        }

    }
}
